lib/transaction.ex

```diff
@@ -142,26 +143,33 @@ defmodule Bitcoinex.Transaction.Utils do
     end
   end
 
-  @doc """
-    Serializes a transaction without the witness structure.
-  """
-  @spec serialize_legacy(Transaction.t()) :: iolist()
-  def serialize_legacy(txn) do
+  @spec serialize(%Transaction{witnesses: nil}) :: binary
```

Comment:
Should this be `@spec serialize(%Transaction{witnesses: nil | list}) :: binary`?